xds/internal/balancer/wrrlocality/balancer_test.go

```diff
@@ -119,3 +130,135 @@ func (s) TestParseConfig(t *testing.T) {
+	weightedTargetName = "mock_weighted_target"
+	stub.Register("mock_weighted_target", stub.BalancerFuncs{
+		ParseConfig: func(c json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
+			var cfg weightedtarget.LBConfig
+			if err := json.Unmarshal(c, &cfg); err != nil {
+				return nil, err
+			}
+			return &cfg, nil
+		},
+		UpdateClientConnState: func(bd *stub.BalancerData, ccs balancer.ClientConnState) error {
+			wtCfg, ok := ccs.BalancerConfig.(*weightedtarget.LBConfig)
+			if !ok {
+				return errors.New("child received config that was not a weighted target config")
+			}
+			defer cfgCh.Send(wtCfg)
+			return nil
+		},
+	})
+
+	wrrL, close := setup(t)
+	defer close()
+
+	// Create the addresses with two localities with certain locality weights.
+	// This represents what the addresses the wrr_locality balancer will receive
+	// in UpdateClientConnState.
+	var addrs []resolver.Address
+	addr := resolver.Address{
+		Addr: "locality-1",
+	}
+
+	lID := internal.LocalityID{
+		Region:  "region-1",
+		Zone:    "zone-1",
+		SubZone: "subzone-1",
+	}
+	addr = internal.SetLocalityID(addr, lID)
+	addr = SetAddrInfo(addr, AddrInfo{LocalityWeight: 2})
+	addrs = append(addrs, addr)
+
+	addr2 := resolver.Address{
+		Addr: "locality-2",
+	}
+	lID2 := internal.LocalityID{
+		Region:  "region-2",
+		Zone:    "zone-2",
+		SubZone: "subzone-2",
+	}
+	addr2 = internal.SetLocalityID(addr2, lID2)
+	addr2 = SetAddrInfo(addr2, AddrInfo{LocalityWeight: 1})
+	addrs = append(addrs, addr2)
+
+	err := wrrL.UpdateClientConnState(balancer.ClientConnState{
+		BalancerConfig: &LBConfig{
+			ChildPolicy: &internalserviceconfig.BalancerConfig{
+				Name: "round_robin",
+			},
+		},
+		ResolverState: resolver.State{
+			Addresses: addrs,
+		},
+	})
+	if err != nil {
+		t.Fatalf("Unexpected error from UpdateClientConnState: %v", err)
+	}
+
+	// Note that these inline strings declared as the key in Targets built from
+	// Locality ID are not exactly what is shown in the example in the gRFC.
+	// However, this is an implementation detail that does not affect
+	// correctness (confirmed with Java team). The important thing is to get
+	// those three pieces of information region, zone, and subzone down to the
+	// child layer.
+	wantWtCfg := &weightedtarget.LBConfig{
+		Targets: map[string]weightedtarget.Target{
+			"{\"region\":\"region-1\",\"zone\":\"zone-1\",\"subZone\":\"subzone-1\"}": {
+				Weight: 2,
+				ChildPolicy: &internalserviceconfig.BalancerConfig{
+					Name: "round_robin",
+				},
+			},
+			"{\"region\":\"region-2\",\"zone\":\"zone-2\",\"subZone\":\"subzone-2\"}": {
+				Weight: 1,
+				ChildPolicy: &internalserviceconfig.BalancerConfig{
+					Name: "round_robin",
+				},
+			},
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	cfg, err := cfgCh.Receive(ctx)
+	if err != nil {
+		// This means UpdateClientConnState on the child never got called.
+		t.Fatalf("Error from cfgOrErrCh: %v", err)
```

Comment:
Please make this error message more useful. When a test fails, the reader of the output should have enough information to understand what it happening even without looking at the test code. This error string does not provide any information to me.

Reply:
Switched to "No signal received from UpdateClientConnState() on the child: %v". Let me know what you think.